Expo1900/Expo1900/Controller/KoreaEntryListViewController.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  KoreaEntryListViewController.swift
+//  Expo1900
+//
+//  Created by MARY, KOBE on 2023/06/30.
+//
+
+import UIKit
+
+final class KoreaEntryListViewController: UIViewController {
+    @IBOutlet weak var entryListTableView: UITableView!
+
+	private let itemsModel: [ItemsModel]? = try? Decoder.decodeJSON(dataAssetName: JSONDataNameSpace.entryData)
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        entryListTableView.dataSource = self
+    }
+	
+	override func viewWillAppear(_ animated: Bool) {
+		super.viewWillAppear(animated)
+        
+		self.title = ViewControllerTitleNameSpace.koreaEntryList
+		navigationController?.setNavigationBarHidden(false, animated: false)
+	}
+	
+	override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+		guard let destination = segue.destination as? EntryDetailViewController,
+			  let selectedIndex = self.entryListTableView.indexPathForSelectedRow?.row,
+              let itemsModel = itemsModel else {
+			return
+		}
```

Comment:
이 부분은 저라면 [공식문서](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621490-prepare) 를 참고해서 구현할 것 같아요 :). 

> Because segues can be triggered from multiple sources, you can use the information in the segue and sender parameters to disambiguate between different logical paths in your app. For example, if the segue originated from a table view, the sender parameter would identify the table view cell that the user tapped. You could then use that information to set the data on the destination view controller. 

혹시 만약 이 부분에 대해서 더 궁금하시다면 언제든 dm남겨주세요 :)

Reply:
- 🙋‍♀️🙋‍♂️  답변

```swift!
 override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        guard let destination = segue.destination as? EntryDetailViewController,
              let indexPath = entryListTableView.indexPath(for: sender as! UITableViewCell),
              let itemsModel = itemsModel else {
            return
        }
        let selectedIndex = indexPath.row
        
        destination.koreanEntryImage = UIImage(named: itemsModel[selectedIndex].imageName)
        destination.entryDescription = itemsModel[selectedIndex].desc
        destination.entryTitle = itemsModel[selectedIndex].name
    }
```
공식문서를 참고하여 위와 같은 코드를 작성해보았습니다. 
sender를 UITableViewCell로 다운캐스팅하여 indexPath를 얻도록 수정하였습니다. 
하지만 결과적으로 `EntryDetailViewController` 초기화 시 데이터를 넘겨주는 방법으로 변경하면서 화면 전환 방법을 변경하였고, 위 코드는 삭제하였습니다.

Reply:
sender와 타입 캐스팅을 활용하면 어렵게 보이는 문제도 보다 쉽게 해결할 때가 종종 있더라구요 :) 👍👍👍